src/hooks/useRegister.js

```diff
@@ -0,0 +1,28 @@
+import { useMutation } from "@tanstack/react-query";
+import { registerUser } from "../api/users";
+import { useNavigate } from "react-router-dom";
+
+export const useRegister = ({ setUserExist, setUser }) => {
```

Comment:
please change the name to userRegisterUser to be more explicit on what are you registering